### PR TITLE
[Dashboard] Make loading screen not block out the entire page.

### PR DESCRIPTION
**Pull request by alanwguo**
## Why are these changes needed?
Previously, if a dashboard page was loading, it would grey out the whole screen and buttons would not be press-able. Now, we don't block out the whole page.
Also don't show loading bar if data is already loaded from in-memory cache.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
Can you share a script of the new loading behavior?

**Comment by alanwguo** [comment]
> Can you share a script of the new loading behavior?


https://user-images.githubusercontent.com/711935/233207229-0fcd01c8-908a-4541-9817-b4d7b50bac34.mov

**Review by rkooo567 — APPROVED** [review]
Maybe it can be done from a follow up, but for each card, is it possible to write the last-refreshed time?

**Comment by chaowanggg** [comment]
lgtm!

**Comment by scottsun94** [comment]
@rkooo567 hmm. I'm still waiting for @alanwguo 's response before giving some visual feedback... We can address it in another pr

**Comment by rkooo567** [comment]
Oops... I will wait for your approval for frontend PR next time :(

**Comment by rkooo567** [comment]
Btw, I merged this since Alan's out, and this will block Chao's future work; https://github.com/ray-project/ray/pull/34514

(I assumed you approved the PR based on your comment, but lmk if I am wrong!)

**Pull request merged**
